pkg/controller/daemon/daemon_controller.go

```diff
@@ -1130,28 +1130,30 @@ func (dsc *DaemonSetsController) updateDaemonSetStatus(ctx context.Context, ds *
 
 		if shouldRun {
 			desiredNumberScheduled++
-			if scheduled {
```

Comment:
You could go one step further with your optimizations, this if condition has an else, which you could turn into:
```golang
		if !shouldRun {
			if scheduled {
				numberMisscheduled++
			}
			continue
		}
```
above this current `if shouldRun {...` condition, this way you'd be able to get rid of yet another indentation level. 

Feel free to open a followup and ping me on slack for review.